src/posix/tty.rs

```diff
@@ -130,6 +167,9 @@ impl TTYPort {
         // other applications that may have an exclusive port lock.
         ioctl::tiocexcl(fd.0)?;
 
+        // Also use flock to lock the port
+        flock_exclusive(fd.0)?;
```

Comment:
After sleeping over it: Shouldn't we make this a "all or nothing" operation and attempt to gracefully retreat from this situation an not leave a port locked in the one way or the other?

Reply:
Right. I should add a note on that for future me.